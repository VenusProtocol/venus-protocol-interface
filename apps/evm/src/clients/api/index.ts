export { default as queryClient } from './queryClient';

// Mutations
export { default as mintVai } from './mutations/mintVai';
export * from './mutations/mintVai';
export { default as useMintVai } from './mutations/mintVai/useMintVai';

export { default as repayVai } from './mutations/repayVai';
export * from './mutations/repayVai';
export { default as useRepayVai } from './mutations/repayVai/useRepayVai';

export { default as enterMarket } from './mutations/enterMarket';
export * from './mutations/enterMarket';
export { default as useEnterMarket } from './mutations/enterMarket/useEnterMarket';

export { default as exitMarket } from './mutations/exitMarket';
export * from './mutations/exitMarket';
export { default as useExitMarket } from './mutations/exitMarket/useExitMarket';

export { default as approveToken } from './mutations/approveToken';
export * from './mutations/approveToken';
export { default as useApproveToken } from './mutations/approveToken/useApproveToken';

export { default as revokeSpendingLimit } from './mutations/revokeSpendingLimit';
export * from './mutations/revokeSpendingLimit';
export { default as useRevokeSpendingLimit } from './mutations/revokeSpendingLimit/useRevokeSpendingLimit';

export { default as supply } from './mutations/supply';
export * from './mutations/supply';
export { default as useSupply } from './mutations/supply/useSupply';
export * from './mutations/supply/useSupply';

export { default as redeem } from './mutations/redeem';
export * from './mutations/redeem';
export { default as useRedeem } from './mutations/redeem/useRedeem';

export { default as repay } from './mutations/repay';
export * from './mutations/repay';
export { default as useRepay } from './mutations/repay/useRepay';

export { default as redeemUnderlying } from './mutations/redeemUnderlying';
export * from './mutations/redeemUnderlying';
export { default as useRedeemUnderlying } from './mutations/redeemUnderlying/useRedeemUnderlying';

export { default as borrow } from './mutations/borrow';
export * from './mutations/borrow';
export { default as useBorrow } from './mutations/borrow/useBorrow';

export { default as withdrawXvs } from './mutations/withdrawXvs';
export * from './mutations/withdrawXvs';
export { default as useWithdrawXvs } from './mutations/withdrawXvs/useWithdrawXvs';

export { default as setVoteDelegate } from './mutations/setVoteDelegate';
export * from './mutations/setVoteDelegate';
export { default as useSetVoteDelegate } from './mutations/setVoteDelegate/useSetVoteDelegate';

export { default as createProposal } from './mutations/createProposal';
export * from './mutations/createProposal';
export { default as useCreateProposal } from './mutations/createProposal/useCreateProposal';

export { default as cancelProposal } from './mutations/cancelProposal';
export * from './mutations/cancelProposal';
export { default as useCancelProposal } from './mutations/cancelProposal/useCancelProposal';

export { default as executeProposal } from './mutations/executeProposal';
export * from './mutations/executeProposal';
export { default as useExecuteProposal } from './mutations/executeProposal/useExecuteProposal';

export { default as queueProposal } from './mutations/queueProposal';
export * from './mutations/queueProposal';
export { default as useQueueProposal } from './mutations/queueProposal/useQueueProposal';

export { default as stakeInXvsVault } from './mutations/stakeInXvsVault';
export * from './mutations/stakeInXvsVault';
export { default as useStakeInXvsVault } from './mutations/stakeInXvsVault/useStakeInXvsVault';

export { default as stakeInVaiVault } from './mutations/stakeInVaiVault';
export * from './mutations/stakeInVaiVault';
export { default as useStakeInVaiVault } from './mutations/stakeInVaiVault/useStakeInVaiVault';

export { default as castVote } from './mutations/vote/castVote';
export * from './mutations/vote/castVote';
export { default as useCastVote } from './mutations/vote/useCastVote';

export { default as castVoteWithReason } from './mutations/vote/castVoteWithReason';
export * from './mutations/vote/castVoteWithReason';
export { default as useCastVoteWithReason } from './mutations/vote/useCastVoteWithReason';

export { default as withdrawFromVaiVault } from './mutations/withdrawFromVaiVault';
export * from './mutations/withdrawFromVaiVault';
export { default as useWithdrawFromVaiVault } from './mutations/withdrawFromVaiVault/useWithdrawFromVaiVault';

export { default as requestWithdrawalFromXvsVault } from './mutations/requestWithdrawalFromXvsVault';
export * from './mutations/requestWithdrawalFromXvsVault';
export { default as useRequestWithdrawalFromXvsVault } from './mutations/requestWithdrawalFromXvsVault/useRequestWithdrawalFromXvsVault';

export { default as executeWithdrawalFromXvsVault } from './mutations/executeWithdrawalFromXvsVault';
export * from './mutations/executeWithdrawalFromXvsVault';
export { default as useExecuteWithdrawalFromXvsVault } from './mutations/executeWithdrawalFromXvsVault/useExecuteWithdrawalFromXvsVault';

export { default as swapTokens } from './mutations/swapTokens';
export * from './mutations/swapTokens';
export { default as useSwapTokens } from './mutations/swapTokens/useSwapTokens';

export { default as swapTokensAndRepay } from './mutations/swapTokensAndRepay';
export * from './mutations/swapTokensAndRepay';
export { default as useSwapTokensAndRepay } from './mutations/swapTokensAndRepay/useSwapTokensAndRepay';

export { default as swapTokensAndSupply } from './mutations/swapTokensAndSupply';
export * from './mutations/swapTokensAndSupply';
export { default as useSwapTokensAndSupply } from './mutations/swapTokensAndSupply/useSwapTokensAndSupply';

export { default as claimRewards } from './mutations/claimRewards';
export * from './mutations/claimRewards';
export { default as useClaimRewards } from './mutations/claimRewards/useClaimRewards';

export { default as useStakeInVault } from './mutations/useStakeInVault';

export { default as claimPrimeToken } from './mutations/claimPrimeToken';
export * from './mutations/claimPrimeToken';
export { default as useClaimPrimeToken } from './mutations/claimPrimeToken/useClaimPrimeToken';

export { default as bridgeXvs } from './mutations/bridgeXvs';
export * from './mutations/bridgeXvs';
export { default as useBridgeXvs } from './mutations/bridgeXvs/useBridgeXvs';

export { default as wrapTokensAndSupply } from './mutations/wrapTokensAndSupply';
export * from './mutations/wrapTokensAndSupply';
export { default as useWrapTokensAndSupply } from './mutations/wrapTokensAndSupply/useWrapTokensAndSupply';

export { default as updatePoolDelegateStatus } from './mutations/updatePoolDelegateStatus';
export * from './mutations/updatePoolDelegateStatus';
export { default as useUpdatePoolDelegateStatus } from './mutations/updatePoolDelegateStatus/useUpdatePoolDelegateStatus';

export { default as borrowAndUnwrap } from './mutations/borrowAndUnwrap';
export * from './mutations/borrowAndUnwrap';
export { default as useBorrowAndUnwrap } from './mutations/borrowAndUnwrap/useBorrowAndUnwrap';

export { default as redeemAndUnwrap } from './mutations/redeemAndUnwrap';
export * from './mutations/redeemAndUnwrap';
export { default as useRedeemAndUnwrap } from './mutations/redeemAndUnwrap/useRedeemAndUnwrap';

export { default as redeemUnderlyingAndUnwrap } from './mutations/redeemUnderlyingAndUnwrap';
export * from './mutations/redeemUnderlyingAndUnwrap';
export { default as useRedeemUnderlyingAndUnwrap } from './mutations/redeemUnderlyingAndUnwrap/useRedeemUnderlyingAndUnwrap';

// Queries
export { default as getVaiCalculateRepayAmount } from './queries/getVaiCalculateRepayAmount';
export * from './queries/getVaiCalculateRepayAmount';
export { default as useGetVaiCalculateRepayAmount } from './queries/getVaiCalculateRepayAmount/useGetVaiCalculateRepayAmount';

export { default as getVaiRepayAmountWithInterests } from './queries/getVaiRepayAmountWithInterests';
export * from './queries/getVaiRepayAmountWithInterests';
export { default as useGetVaiRepayAmountWithInterests } from './queries/getVaiRepayAmountWithInterests/useGetVaiRepayAmountWithInterests';

export { default as getVaiTreasuryPercentage } from './queries/getVaiTreasuryPercentage';
export * from './queries/getVaiTreasuryPercentage';
export { default as useGetVaiTreasuryPercentage } from './queries/getVaiTreasuryPercentage/useGetVaiTreasuryPercentage';

export { default as getHypotheticalAccountLiquidity } from './queries/getHypotheticalAccountLiquidity';
export * from './queries/getHypotheticalAccountLiquidity';

export { default as getLegacyPoolMarkets } from './queries/getLegacyPoolMarkets';
export * from './queries/getLegacyPoolMarkets';
export { default as useGetLegacyPoolMarkets } from './queries/getLegacyPoolMarkets/useGetLegacyPoolMarkets';

export { default as getVTokenBalancesAll } from './queries/getVTokenBalancesAll';
export * from './queries/getVTokenBalancesAll';
export { default as useGetVTokenBalancesAll } from './queries/getVTokenBalancesAll/useGetVTokenBalancesAll';

export { default as getVTokenBalanceOf } from './queries/getVTokenBalanceOf';
export * from './queries/getVTokenBalanceOf';
export { default as useGetVTokenBalanceOf } from './queries/getVTokenBalanceOf/useGetVTokenBalanceOf';

export { default as getPendingRewards } from './queries/getPendingRewards';
export * from './queries/getPendingRewards';
export { default as useGetPendingRewards } from './queries/getPendingRewards/useGetPendingRewards';

export { default as getIsAddressAuthorized } from './queries/getIsAddressAuthorized';
export * from './queries/getIsAddressAuthorized';
export { default as useGetIsAddressAuthorized } from './queries/getIsAddressAuthorized/useGetIsAddressAuthorized';

export { default as getAllowance } from './queries/getAllowance';
export * from './queries/getAllowance';
export { default as useGetAllowance } from './queries/getAllowance/useGetAllowance';

export { default as getBalanceOf } from './queries/getBalanceOf';
export * from './queries/getBalanceOf';
export { default as useGetBalanceOf } from './queries/getBalanceOf/useGetBalanceOf';

export { default as getTokenBalances } from './queries/getTokenBalances';
export * from './queries/getTokenBalances';
export { default as useGetTokenBalances } from './queries/getTokenBalances/useGetTokenBalances';

export { default as getTokenUsdPrice } from './queries/getTokenUsdPrice';
export * from './queries/getTokenUsdPrice';
export { default as useGetTokenUsdPrice } from './queries/getTokenUsdPrice/useGetTokenUsdPrice';

export { default as getVrtConversionEndTime } from './queries/getVrtConversionEndTime';
export * from './queries/getVrtConversionEndTime';
export { default as useGetVrtConversionEndTime } from './queries/getVrtConversionEndTime/useGetVrtConversionEndTime';

export { default as getVrtConversionRatio } from './queries/getVrtConversionRatio';
export * from './queries/getVrtConversionRatio';
export { default as useGetVrtConversionRatio } from './queries/getVrtConversionRatio/useGetVrtConversionRatio';

export { default as getVenusVaiVaultDailyRate } from './queries/getVenusVaiVaultDailyRate';
export * from './queries/getVenusVaiVaultDailyRate';
export { default as useGetVenusVaiVaultDailyRate } from './queries/getVenusVaiVaultDailyRate/useGetVenusVaiVaultDailyRate';

export { default as getXvsWithdrawableAmount } from './queries/getXvsWithdrawableAmount';
export * from './queries/getXvsWithdrawableAmount';

export { default as useGetXvsWithdrawableAmount } from './queries/getXvsWithdrawableAmount/useGetXvsWithdrawableAmount';

export { default as useGetAsset } from './queries/useGetAsset';

export { default as useGetLegacyPoolTotalXvsDistributed } from './queries/useGetLegacyPoolTotalXvsDistributed';

export * from './queries/getLegacyPool';
export { default as useGetLegacyPool } from './queries/getLegacyPool/useGetLegacyPool';

export { default as useGetPools } from './queries/useGetPools';

export { default as useGetPool } from './queries/useGetPool';

export { default as useGetIsolatedPoolsTreasuryTotals } from './queries/useGetIsolatedPoolsTreasuryTotals';

export { default as getIsolatedPools } from './queries/getIsolatedPools';
export * from './queries/getIsolatedPools';
export { default as useGetIsolatedPools } from './queries/getIsolatedPools/useGetIsolatedPools';

export { default as getMarketHistory } from './queries/getMarketHistory';
export * from './queries/getMarketHistory';
export { default as useGetMarketHistory } from './queries/getMarketHistory/useGetMarketHistory';

export { default as getVTokenInterestRateModel } from './queries/getVTokenInterestRateModel';
export * from './queries/getVTokenInterestRateModel';
export { default as useGetVTokenInterestRateModel } from './queries/getVTokenInterestRateModel/useGetVTokenInterestRateModel';

export { default as getVTokenApySimulations } from './queries/getVTokenApySimulations';
export * from './queries/getVTokenApySimulations';
export { default as useGetVTokenApySimulations } from './queries/getVTokenApySimulations/useGetVTokenApySimulations';

export { default as getCurrentVotes } from './queries/getCurrentVotes';
export * from './queries/getCurrentVotes';
export { default as useGetCurrentVotes } from './queries/getCurrentVotes/useGetCurrentVotes';

export { default as getTransactions } from './queries/getTransactions';
export * from './queries/getTransactions';
export { default as useGetTransactions } from './queries/getTransactions/useGetTransactions';

export { default as getXvsVaultPoolCount } from './queries/getXvsVaultPoolCount';
export * from './queries/getXvsVaultPoolCount';
export { default as useGetXvsVaultPoolCount } from './queries/getXvsVaultPoolCount/useGetXvsVaultPoolCount';

export { default as getXvsVaultPoolInfo } from './queries/getXvsVaultPoolInfo';
export * from './queries/getXvsVaultPoolInfo';
export { default as useGetXvsVaultPoolInfo } from './queries/getXvsVaultPoolInfo/useGetXvsVaultPoolInfo';

export { default as getXvsVaultRewardPerBlock } from './queries/getXvsVaultRewardPerBlock';
export * from './queries/getXvsVaultRewardPerBlock';
export { default as useGetXvsVaultRewardPerBlock } from './queries/getXvsVaultRewardPerBlock/useGetXvsVaultRewardPerBlock';

export { default as getXvsVaultTotalAllocationPoints } from './queries/getXvsVaultTotalAllocationPoints';
export * from './queries/getXvsVaultTotalAllocationPoints';
export { default as useGetXvsVaultTotalAllocationPoints } from './queries/getXvsVaultTotalAllocationPoints/useGetXvsVaultTotalAllocationPoints';

export { default as getXvsVaultUserInfo } from './queries/getXvsVaultUserInfo';
export * from './queries/getXvsVaultUserInfo';
export { default as useGetXvsVaultUserInfo } from './queries/getXvsVaultUserInfo/useGetXvsVaultUserInfo';

export { default as getXvsVaultLockedDeposits } from './queries/getXvsVaultLockedDeposits';
export * from './queries/getXvsVaultLockedDeposits';
export { default as useGetXvsVaultLockedDeposits } from './queries/getXvsVaultLockedDeposits/useGetXvsVaultLockedDeposits';

export { default as useGetVaults } from './queries/useGetVaults';

export { default as getProposals } from './queries/getProposals';
export * from './queries/getProposals';
export { default as useGetProposals } from './queries/getProposals/useGetProposals';

export { default as getProposal } from './queries/getProposals/getProposal';
export * from './queries/getProposals/getProposal';
export { default as useGetProposal } from './queries/getProposals/useGetProposal';

export { default as getVoteReceipt } from './queries/getVoteReceipt';
export * from './queries/getVoteReceipt';
export { default as useGetVoteReceipt } from './queries/getVoteReceipt/useGetVoteReceipt';

export { default as getVoters } from './queries/getVoters';
export * from './queries/getVoters';
export { default as useGetVoters } from './queries/getVoters/useGetVoters';

export { default as getVoterDetails } from './queries/getVoterDetails';
export * from './queries/getVoterDetails';
export { default as useGetVoterDetails } from './queries/getVoterDetails/useGetVoterDetails';

export { default as getVoteSummary } from './queries/getVoteSummary';
export * from './queries/getVoteSummary';
export { default as useGetVoteSummary } from './queries/getVoteSummary/useGetVoteSummary';

export { default as getVoterHistory } from './queries/getVoterHistory';
export * from './queries/getVoterHistory';
export { default as useGetVoterHistory } from './queries/getVoterHistory/useGetVoterHistory';

export { default as getVaiVaultUserInfo } from './queries/getVaiVaultUserInfo';
export * from './queries/getVaiVaultUserInfo';
export { default as useGetVaiVaultUserInfo } from './queries/getVaiVaultUserInfo/useGetVaiVaultUserInfo';

export { default as useGetVestingVaults } from './queries/useGetVaults/useGetVestingVaults';

export { default as getVoteDelegateAddress } from './queries/getVoteDelegateAddress';
export * from './queries/getVoteDelegateAddress';
export { default as useGetVoteDelegateAddress } from './queries/getVoteDelegateAddress/useGetVoteDelegateAddress';

export { default as getVoterAccounts } from './queries/getVoterAccounts';
export * from './queries/getVoterAccounts';
export { default as useGetVoterAccounts } from './queries/getVoterAccounts/useGetVoterAccounts';

export { default as getProposalThreshold } from './queries/getProposalThreshold';
export * from './queries/getProposalThreshold';
export { default as useGetProposalThreshold } from './queries/getProposalThreshold/useGetProposalThreshold';

export { default as getProposalState } from './queries/getProposalState';
export * from './queries/getProposalState';
export { default as useGetProposalState } from './queries/getProposalState/useGetProposalState';

export { default as getLatestProposalIdByProposer } from './queries/getLatestProposalIdByProposer';
export * from './queries/getLatestProposalIdByProposer';
export { default as useGetLatestProposalIdByProposer } from './queries/getLatestProposalIdByProposer/useGetLatestProposalIdByProposer';

export { default as getMintableVai } from './queries/getMintableVai';
export * from './queries/getMintableVai';
export { default as useGetMintableVai } from './queries/getMintableVai/useGetMintableVai';

export { default as getBlockNumber } from './queries/getBlockNumber';
export * from './queries/getBlockNumber';
export { default as useGetBlockNumber } from './queries/getBlockNumber/useGetBlockNumber';

export { default as getProposalEta } from './queries/getProposalEta';
export * from './queries/getProposalEta';
export { default as useGetProposalEta } from './queries/getProposalEta/useGetProposalEta';

export { default as getPancakeSwapPairs } from './queries/getPancakeSwapPairs';
export * from './queries/getPancakeSwapPairs';
export { default as useGetPancakeSwapPairs } from './queries/getPancakeSwapPairs/useGetPancakeSwapPairs';

export { default as getVaiRepayApr } from './queries/getVaiRepayApr';
export * from './queries/getVaiRepayApr';
export { default as useGetVaiRepayApr } from './queries/getVaiRepayApr/useGetVaiRepayApr';

export { default as getXvsVaultPendingWithdrawalsFromBeforeUpgrade } from './queries/getXvsVaultPendingWithdrawalsFromBeforeUpgrade';
export * from './queries/getXvsVaultPendingWithdrawalsFromBeforeUpgrade';
export { default as useGetXvsVaultPendingWithdrawalsFromBeforeUpgrade } from './queries/getXvsVaultPendingWithdrawalsFromBeforeUpgrade/useGetXvsVaultPendingWithdrawalsFromBeforeUpgrade';

export { default as getVTokens } from './queries/getVTokens';
export * from './queries/getVTokens';
export { default as useGetVTokens } from './queries/getVTokens/useGetVTokens';

export { default as getPrimeToken } from './queries/getPrimeToken';
export * from './queries/getPrimeToken';
export { default as useGetPrimeToken } from './queries/getPrimeToken/useGetPrimeToken';

export { default as getHypotheticalPrimeApys } from './queries/getHypotheticalPrimeApys';
export * from './queries/getHypotheticalPrimeApys';
export { default as useGetHypotheticalPrimeApys } from './queries/getHypotheticalPrimeApys/useGetHypotheticalPrimeApys';

export { default as getPrimeStatus } from './queries/getPrimeStatus';
export * from './queries/getPrimeStatus';
export { default as useGetPrimeStatus } from './queries/getPrimeStatus/useGetPrimeStatus';

export * from './queries/getLatestAppVersion';
export { default as useGetLatestAppVersion } from './queries/getLatestAppVersion/useGetLatestAppVersion';

export { default as useGetPrimeEstimation } from './queries/useGetPrimeEstimation';

export { default as getPrimeDistributionForMarket } from './queries/getPrimeDistributionForMarket';
export * from './queries/getPrimeDistributionForMarket';
export { default as useGetPrimeDistributionForMarket } from './queries/getPrimeDistributionForMarket/useGetPrimeDistributionForMarket';

export * from './queries/getVaiVaultPaused';
export { default as useGetVaiVaultPaused } from './queries/getVaiVaultPaused/useGetVaiVaultPaused';

export * from './queries/getXvsVaultPaused';
export { default as useGetXvsVaultPaused } from './queries/getXvsVaultPaused/useGetXvsVaultPaused';

export { default as getXvsBridgeFeeEstimation } from './queries/getXvsBridgeFeeEstimation';
export * from './queries/getXvsBridgeFeeEstimation';
export { default as useGetXvsBridgeFeeEstimation } from './queries/getXvsBridgeFeeEstimation/useGetXvsBridgeFeeEstimation';

export { default as getXvsBridgeStatus } from './queries/getXvsBridgeStatus';
export * from './queries/getXvsBridgeStatus';
export { default as useGetXvsBridgeStatus } from './queries/getXvsBridgeStatus/useGetXvsBridgeStatus';

export { default as getXvsBridgeMintStatus } from './queries/getXvsBridgeMintStatus';
export * from './queries/getXvsBridgeMintStatus';
export { default as useGetXvsBridgeMintStatus } from './queries/getXvsBridgeMintStatus/useGetXvsBridgeMintStatus';

export { default as getPoolDelegateApprovalStatus } from './queries/getPoolDelegateApprovalStatus';
export * from './queries/getPoolDelegateApprovalStatus';
export { default as useGetPoolDelegateApprovalStatus } from './queries/getPoolDelegateApprovalStatus/useGetPoolDelegateApprovalStatus';
