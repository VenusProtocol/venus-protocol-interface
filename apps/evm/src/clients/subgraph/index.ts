export * from './queries/isolatedPools/getIsolatedPoolParticipantsCount';
export * from './queries/governance/getProposals';
