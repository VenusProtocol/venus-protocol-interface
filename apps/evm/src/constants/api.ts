import type { Network } from 'types';

export const apiUrls: Record<Network, string> = {
  testnet: 'https://testnetapi.venus.io',
  mainnet: 'https://api.venus.io',
  'mainnet-preview': 'https://api-preview.venus.io',
};
