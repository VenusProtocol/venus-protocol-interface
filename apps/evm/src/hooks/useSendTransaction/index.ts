import { ContractReceipt, ContractTransaction } from 'ethers';
import { MutationObserverOptions, useMutation } from 'react-query';

import FunctionKey from 'constants/functionKey';
import { TransactionType } from 'types';

import { CONFIRMATIONS, useTrackTransaction } from './useTrackTransaction';

export interface UseSendTransactionOptions<TMutateInput extends Record<string, unknown> | void>
  extends MutationObserverOptions<unknown, Error, TMutateInput> {
  waitForConfirmation?: boolean;
}

export interface UseSendTransactionInput<TMutateInput extends Record<string, unknown> | void> {
  fn: (input: TMutateInput) => Promise<ContractTransaction>;
  fnKey: FunctionKey | [FunctionKey, ...unknown[]];
  transactionType?: TransactionType;
  onConfirmed?: (input: {
    transaction: ContractTransaction;
    transactionReceipt: ContractReceipt;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  onReverted?: (input: {
    transaction: ContractTransaction;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  options?: UseSendTransactionOptions<TMutateInput>;
}

export const useSendTransaction = <TMutateInput extends Record<string, unknown> | void>({
  fn,
  fnKey,
  transactionType,
  onConfirmed,
  onReverted,
  options,
}: UseSendTransactionInput<TMutateInput>) => {
  const trackTransaction = useTrackTransaction({ transactionType });

  return useMutation(
    fnKey,
    async input => {
      // Send transaction
      const transaction = await fn(input);

      // Track transaction's progress in the background
      trackTransaction({
        transaction,
        onConfirmed: onConfirmedInput => onConfirmed?.({ ...onConfirmedInput, input }),
        onReverted: onRevertedInput => onReverted?.({ ...onRevertedInput, input }),
      });

      if (options?.waitForConfirmation) {
        // Only return when transaction has been confirmed
        await transaction.wait(CONFIRMATIONS);
      }
    },
    options,
  );
};
