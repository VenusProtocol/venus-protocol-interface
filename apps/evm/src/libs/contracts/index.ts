export * from './generated/addresses';
export * from './generated/abis';

export * from './utilities/getContractAddress';
