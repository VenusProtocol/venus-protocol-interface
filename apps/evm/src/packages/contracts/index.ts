// Export contract getter functions and hooks
export * from './generated/getters';
export * from './hooks/useGetTokenContract';
export * from './hooks/useGetVTokenContract';
export * from './utilities/getTokenContract';
export * from './utilities/getVTokenContract';
// Export types
export * from './generated/infos/contractTypes';
export * from './generated/infos/types';
