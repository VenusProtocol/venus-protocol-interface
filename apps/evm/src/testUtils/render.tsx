import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render as renderComponentTl } from '@testing-library/react';
import { renderHook as renderHookTl } from '@testing-library/react-hooks';
import type { ReactElement } from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import type Vi from 'vitest';

import fakeSigner from '__mocks__/models/signer';

import { Web3Wrapper, useAccountAddress, useChainId, useSigner } from 'libs/wallet';
import { MuiThemeProvider } from 'theme/MuiThemeProvider';
import type { ChainId } from 'types';

const createQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        gcTime: 0,
      },
    },
  });

interface Options {
  accountAddress?: string;
  chainId?: ChainId;
  routerInitialEntries?: string[];
  routePath?: string;
}

interface WrapperProps {
  queryClient: QueryClient;
  children?: React.ReactNode;
  options?: Partial<Options>;
}

const Wrapper: React.FC<WrapperProps> = ({ children, queryClient, options }) => {
  if (options?.accountAddress) {
    const accountAddress = options?.accountAddress;

    (useAccountAddress as Vi.Mock).mockImplementation(() => ({
      accountAddress,
    }));

    (useSigner as Vi.Mock).mockImplementation(() => ({
      signer: {
        ...fakeSigner,
        getAddress: async () => accountAddress,
      },
    }));
  }

  if (options?.chainId) {
    (useChainId as Vi.Mock).mockImplementation(() => ({
      chainId: options?.chainId,
    }));
  }

  return (
    <MuiThemeProvider>
      <QueryClientProvider client={queryClient}>
        <Web3Wrapper>
          <MemoryRouter initialEntries={options?.routerInitialEntries || ['/']}>
            <Routes>
              <Route path={options?.routePath || '/'} element={children} />
            </Routes>
          </MemoryRouter>
        </Web3Wrapper>
      </QueryClientProvider>
    </MuiThemeProvider>
  );
};

export const renderComponent = (children: ReactElement, options?: Partial<Options>) => {
  const queryClient = createQueryClient();

  const renderRes = renderComponentTl(children, {
    wrapper: props => <Wrapper queryClient={queryClient} options={options} {...props} />,
  });

  return {
    ...renderRes,
    queryClient,
  };
};

export const renderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  options?: Partial<Options>,
) => {
  const queryClient = createQueryClient();

  const renderRes = renderHookTl(hook, {
    wrapper: props => <Wrapper queryClient={queryClient} options={options} {...props} />,
  });

  return {
    ...renderRes,
    queryClient,
  };
};
