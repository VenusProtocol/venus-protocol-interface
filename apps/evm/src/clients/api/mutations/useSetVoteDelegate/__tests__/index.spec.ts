import fakeAccountAddress, {
  altAddress as fakeXvsVaultContractAddress,
} from '__mocks__/models/address';
import { queryClient } from 'clients/api';
import { useSendTransaction } from 'hooks/useSendTransaction';
import { useGetXvsVaultContractAddress } from 'libs/contracts';
import { renderHook } from 'testUtils/render';
import type { Mock } from 'vitest';
import { useSetVoteDelegate } from '..';

vi.mock('libs/contracts');

const fakeInput = {
  delegateAddress: '0x123' as const,
};

const fakeOptions = {
  gasless: true,
  waitForConfirmation: true,
};

describe('useSetVoteDelegate', () => {
  beforeEach(() => {
    (useGetXvsVaultContractAddress as Mock).mockReturnValue(fakeXvsVaultContractAddress);
  });

  it('calls useSendTransaction with the correct parameters', async () => {
    renderHook(() => useSetVoteDelegate(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(fakeInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "address": "0xa258a693A403b7e98fd05EE9e1558C760308cFC7",
        "args": [
          "0x123",
        ],
        "functionName": "delegate",
      }
    `,
    );

    onConfirmed({ input: fakeInput });

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('throws when contract address could not be retrieved', async () => {
    (useGetXvsVaultContractAddress as Mock).mockReturnValue(undefined);

    renderHook(() => useSetVoteDelegate(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const { fn } = (useSendTransaction as Mock).mock.calls[0][0];

    await expect(async () => fn(fakeInput)).rejects.toThrow('somethingWentWrong');
  });
});
