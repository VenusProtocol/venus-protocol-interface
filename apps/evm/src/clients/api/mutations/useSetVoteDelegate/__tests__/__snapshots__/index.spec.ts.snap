// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useSetVoteDelegate > calls useSendTransaction with the correct parameters 2`] = `
[
  [
    {
      "queryKey": [
        "GET_VOTE_DELEGATE_ADDRESS",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_CURRENT_VOTES",
        "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
      ],
    },
  ],
]
`;
